modules/gapi/src/streaming/gstreamer/gstreamersource.cpp

```diff
@@ -184,10 +185,27 @@ void GStreamerSource::Priv::prepareVideoMeta()
             cv::util::throw_error(std::logic_error("Cannot query video width/height."));
         }
 
+        // Fill GstVideoInfo structure to work further with GstVideoFrame class.
+        if (!gst_video_info_from_caps(&m_videoInfo, prerollCaps)) {
+            cv::util::throw_error(std::logic_error("preroll sample has invalid caps."));
+        }
+        m_type = GST_VIDEO_INFO_FORMAT(&m_videoInfo);
         switch(m_outputType) {
             case GStreamerSource::OutputType::FRAME: {
                 // Construct metadata for media frame.
-                m_mediaFrameMeta = GFrameDesc { cv::MediaFormat::NV12, cv::Size(width, height) };
+                switch (m_type) {
+                    case GST_VIDEO_FORMAT_NV12: {
+                        m_mediaFrameMeta = GFrameDesc{ cv::MediaFormat::NV12, cv::Size(width, height) };
+                        break;
+                    }
+                    case GST_VIDEO_FORMAT_GRAY8: {
+                        m_mediaFrameMeta = GFrameDesc{ cv::MediaFormat::GRAY, cv::Size(width, height) };
```

Comment:
so is it `GRAY` or `GRAY8`? I believe I've seen `GRAY8` somewhere in the doxygen comments (and I believe it is quite legit)

Reply:
There are different Gst gray formats. GRAY8 GRAY16. MediaFormat GRAY assumes 8-bit integer numbers. Should we align our naming approach with Gst. I have some doubts that 16 bit will be asked by someone.

Reply:
> I have some doubts that 16 bit will be asked by someone.

It seem to be useful for some scenarios, see #18694. Furthermore, there are two variants of this format: GRAY16_LE and GRAY16_BE